### DISCUSSION TITLE
Variable Density Jet DNS

### DISCUSSION THREAD

**Discussion by ayush-9s** [xcompact3d/Incompact3d#144] [Q&A] [answered]
Hello,
I am trying to perform DNS for variable density jet flows, and based on the non-dimensionalization I am using, I had to modify the calc_eos subroutines and divu_constraint subroutine accordingly. I have specified the desired inflow velocity and temperature distributions in the BC-Jet.f90 file. 

When I try to run the simulation, the velocity at the inflow boundary does not match the boundary condition specified, and it changes linearly with the jet(dens2) to ambient density(dens1) ratio at the inlet. The pressure distribution obtained at all time steps seems incorrect, with a steep change at the inlet at the interface of the ambient flow and the jet(attached the pressure distribution at the inlet plane where y=0 along the x=5 line: x-axis - z, y-axis - pressure). After a few time steps, the max divergence of U diverges and the simulation is terminated. 
I removed all the changes made to the calc_eos and divu_constraint subroutines and again tried to run the simulation but it still showed similar problems. 

Is there something I am missing here? 
I have attached the modified BC-Jet.f90, tools.f90, navier.f90 and the input.i3d files here for reference.
Any help will be appreciated.
Thanks,
Ayush.

[BC-Jet.f90.txt](https://github.com/xcompact3d/Incompact3d/files/10719178/BC-Jet.f90.txt)
[navier.f90.txt](https://github.com/xcompact3d/Incompact3d/files/10719180/navier.f90.txt)
[tools.f90.txt](https://github.com/xcompact3d/Incompact3d/files/10719181/tools.f90.txt)
[input.i3d.txt](https://github.com/xcompact3d/Incompact3d/files/10719237/input.i3d.txt)
![pp](https://user-images.githubusercontent.com/85876808/218384366-5186a97e-323a-4d93-b0e3-24f8cb40d3b0.png)

**Comment by pbartholomew08** [comment, marked as answer]
Hi Ayush,

The jet case is a work in progress and so should not be on the main repository. I would need to look into the BC file for the inlet boundary effect you are observing, from memory it is trying to maintain a fixed mass flow rate, rather than a velocity, however the profile you have sent suggests there is still a bug here.

I apologise for the confusion that this has caused by being in the main repository, we will remove it for now and hopefully restore the case in the future.

P
